api-get-object.go

```diff
@@ -289,6 +292,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, syscall.EBADF
```

Comment:
@harshavardhana It doesn't make sense to return `NoSuchKey`. IMO `InvalidObjectState`[1] is a more apt error response code. thoughts? 

[1] http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList